### PR TITLE
Composition parser

### PR DESCRIPTION
**Pull request by kolayne**
Implemented a module to parse compositions from text. It includes:
- TOP Parser - Transducers-based One-dimensional Polymorphic Parser
- TxtCompositionParser - the module to parse compositions (works on top of TOP)
- A sample txt file that is successfully parsed into a composition

### CODE REVIEW

File: HaskellMusicEngine.cabal [hunk 1]
```diff
@@ -38,5 +40,6 @@ executable HaskellMusicEngine
     ghc-options: -Wall -fno-warn-type-defaults -threaded -rtsopts -with-rtsopts=-N
     build-depends:
          base >=4.7 && < 5
+      ,  strings == 1.1
```
Comment (ilnarkhasanov) [inline HaskellMusicEngine.cabal:43]: Is it really good to have '=='?

File: app/TxtCompositionParser.hs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+-- | Try to parse the given string to a composition.
+-- | On success return `Right composition`. On failure return the tokens that
+-- | were not successfully parsed.
+parseComposition :: String -> Either ([String], Maybe String) Composition
+parseComposition s =
+    let pretokens = pretokenize s in
+        case TOP.parse pretokens Init isFinal transition of
+             Left err  -> Left err
+             Right out -> Right $ Composition out
```
Comment (ilnarkhasanov) [inline app/TxtCompositionParser.hs:118]: $ is beautiful, but not so useful as ()

**Review by ilnarkhasanov — COMMENTED** [review]
LGTM!

**Review by ilnarkhasanov — APPROVED** [review]
LGTM

**Review by Probirochniy — APPROVED** [review]

**Pull request merged**